Dp 9

ItsRizee opened the pull request:
Added: product page
Added: theme toggle
Updated: catalog page
Updated: new product drawer
Updated: theme colors

ItsRizee committed:
Added: Add product drawer
Added: 404 and 500 page
Added: global user state with zustand
Added: Standart layout for all pages

ItsRizee committed:
Added: Add product drawer
Added: 404 and 500 page
Added: global user state with zustand
Added: Standart layout for all pages

ItsRizee committed:
Fixed: Navbar rendering
Fixed: listeners for user's data change

ItsRizee committed:
Added: Catalog
Added: Displaying craftsman on home page
Added: About us page
Updated: removed static info from Product card and Avatar icon

ItsRizee committed:
Added: Product page
Added: owner, tags and createdDate fields to product class
Fixed: rendering of catalog page
Updated: Product card's details and made it a button

ItsRizee committed:
Merge remote-tracking branch 'origin/main'

ItsRizee committed:
Product page update
Added: Input fields for tags and timeline
Added: Image, details and timeline to product page
Added: theme selection in the nav bar
Updated: product card's design and details
Updated: catalog page

ItsRizee committed:
Fixing bugs
Fixed: theme toggle
Fixed: design of website
Fixed: fitting of images in product card
Removed: unused theme attribute in database and user object

atanasovstefan commented on diploma-project/components/DropFileZone.jsx:
```diff
@@ -1,18 +1,70 @@
+        event.preventDefault();
+        setIsDragging(false);
+
+        const file = event.dataTransfer.files[0];
```
The handleDrop and handleChange functions currently accept any type of file. Depending on your specific use case, you might want to add some checks to ensure that the dropped or selected file is of the correct type (e.g., an image). Although you have specified the accept attribute in the input field, it's still a good practice to validate the file type in your JavaScript code.

ItsRizee replied:
Its okay if I make a basic validation of the file extension in my JavaScript code, right?

atanasovstefan replied:
Yep, that should be enough

atanasovstefan commented on diploma-project/components/InputFieldTags.jsx:
```diff
@@ -0,0 +1,62 @@
+                    onChange={handleTagChange}
+                    className="input input-bordered w-full pr-24"
+                />
+                <button type="button" className="btn btn-ghost absolute right-5" onClick={handleAddTag}>
```
The handleAddTag function is only called when the button is clicked. You might want to allow the user to add a tag by pressing the Enter key as well. You can do this by adding an `onKeyPress` event handler to the input field.

ItsRizee replied:
`onKeyPress` is deprecated, but I will see what I can do about this suggestion.

atanasovstefan replied:
Ah, yeah, I keep remembering old stuff. You can leave this for later, but it's something that is easily noticeable

atanasovstefan commented on diploma-project/components/InputFieldTimeline.jsx:
```diff
@@ -0,0 +1,79 @@
+                                        placeholder=""
+                                        value={section.heading}
+                                        onChange={(e) => {
+                                            setTimeline([...timeline.slice(0, index), {
```
This handler creates a new timeline array every time the user types a character. This could be inefficient if the timeline has a lot of sections. You might want to consider using a different state management strategy, such as using a separate state variable for each section.

atanasovstefan commented on diploma-project/components/ProductCard.jsx:
```diff
@@ -1,33 +1,91 @@
+    const router = useRouter();
+    const tags = ["silver craft", "wood craftttttttttt"];
+
+    useEffect(() => {
```
The getUserById function is called in the useEffect hook without any dependencies. This means it will only be called once when the component mounts. If the product.owner prop changes, the user won't be fetched again. Consider if you want to add product.owner to the dependencies array of the useEffect hook, so that it can update.

ItsRizee replied:
I don't think an owner of a product should change. That is why I haven't added any dependencies to the useEffect hook.

atanasovstefan commented on diploma-project/pages/catalog/[id].jsx:
```diff
@@ -0,0 +1,243 @@
+    const [products, setProducts] = useState(null);
+    const router = useRouter();
+
+    useEffect(() => {
```
ditto. Depending on your use case, if you want to refetch the catalog when user.catalog changes, you should call getCatalog with user.catalog as an argument.

ItsRizee replied:
I have `user.catalog` as a dependency in my `useEffect` so I will refetch the catalog if there are changes. There is a difference between `user.catalog` and `catalog` variables. The `user.catalog` is a list of IDs whereas `catalog` is a list of products. I will rename it so it becomes more clear - `catalog` to `products`.

ItsRizee committed:
Fixing bugs
Added: adding tags on enter
Added: check of file type with javascript
Fixed: labels for all input fields
